src/App.tsx

```diff
@@ -1,17 +1,122 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { TodoList } from './TodoList';
 
 import users from './api/users';
+import todos from './api/todos';
+
+const preparesTodos = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId) ?? null,
+}));
 
 const App: React.FC = () => {
+  const [user, setSelectedUser] = useState('');
+  const [todoses, setTodos] = useState([...preparesTodos]);
+  const [title, setTitle] = useState('');
+  const [validTitle, setValidTitle] = useState(false);
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [validUser, setValidUser] = useState(false);
```

Comment:
you can remove `validTitle` and `validUser` variables and use only `errorTitle` and `errorUser`(`!errorTitle` and `!errorUser` when required)